src/components/CalendarMonth.jsx

```diff
@@ -109,7 +109,7 @@ class CalendarMonth extends React.PureComponent {
     this.setMonthTitleHeightTimeout = setTimeout(this.setMonthTitleHeight, 0);
   }
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
```

Comment:
if we're doing the breaking change, we might as well migrate to getDerivedStateFromProps where we can? It's ok to keep the unsafe versions where that's not easy tho.

Reply:
I have a few thoughts on this:
- I want to rock the boat with this PR as little as possible. If people are upgrading from an older react to 16.3+, then upgrading this, if they have a bug it would be nice to say fairly confidently that it wasn't this and probably part of some other part of their migration.
- Related, a _ton_ of the logic in these components is in these lifecycle methods, so I see any substantive change to them as a larger-scope change.
- From my understanding, these methods aren't going anywhere anytime soon. My understanding is the react team has great interest in _discouraging_ their use rather than removing them.
- If you think this is the time/PR to do this lmk and I can see what I can do. Might be more appropriate to do it in two parts.
- Depending on what the future holds, this lib might go the way of many of the other libs out there and get a functional, hooks-based refactor. Maybe instead of migrating all this huge lifecycle method logic twice we just wait until that day comes (though I believe the migration to `getDerivedStateFromProps` is not nearly on the same scale as a migration to hooks, which would be..._non-trivial_.)

I will say in general I'm all about the "while we're at it" mentality. If it were my own repo or wasn't 427,576 weekly downloads then I'd feel more comfortable just doing it 😂 .

My vote is to do it this dumb, rename-only way first to get the warnings out and get the compatibility there, then either have a follow-up PR or include this in future refactoring conversations/planning.

Reply:
If we're going to migrate to hooks any time soon, we'd need to require React v16.9 now, and not just v16.3, even if we do this PR as-is first, and change other things later.

Reply:
Yeah definitely not advocating hooks migration in the near-term.